tests/ConcurrencyTest.php

```diff
@@ -0,0 +1,101 @@
+<?php
```

Comment:
I think this needs to still be an integration test. We need to ensure that we're actually able to send requests concurrently and the handler bindings actually work. I'd feel much more confident in the implementation if we still sent real requests for these basic tests.